### PR TITLE
Add ability to add custom selectors to pdbs and deploys

### PR DESCRIPTION
**Pull request by dblackdblack**
**Please provide a description of this PR:**

It is possible within the gateway helm chart to customize the set of labels applied to pods via the deploy object, but it is not possible to customize the selector labels for the pdb and deploy objects.  This change adds an ability to add additional selector labels to the pdb and deploy objects.

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
Why do we need this when we already have `selectorLabels`? Then entire point is to provide the labels for these selectors

**Comment by dblackdblack** [comment]
I don't see a field selectorLabels in values.yaml:
https://github.com/istio/istio/blob/master/manifests/charts/gateway/values.yaml

I'm doing an in place upgrade from istio 1.16 => 1.19.  The labels on the
gateway pods for both v1.16 and v1.19 share the labels
`app=istioingressgateway` and `istio=ingresgatewayexternal`.  They
additionally have `istio.io/rev=default` (for 1.16) and `
istio.io/rev=1-19-9` (for v1.19).

We manage the k8s service object for gateways outside of helm so that we
can tweak its selector labels easily.

During cutover from 1.16 => 1.19, we will have only the `app` and `istio`
selectors on the service (we will omit `istio.io/rev`), which will match
both v1.16 and v1.19 pods for 50/50 traffic split.

When we're done testing, we'll add the label `istio.io/rev=1-19-0` to the
service selector to target just v1.19.

Regardless, we need the pdb and the deploy objects to have the istio.io/rev
selector the whole time.

**Comment by howardjohn** [comment]
`selectorLabels` is derived from a subset of the labels. Selectors are immutable in K8s deployment so they are intentionally not directly configurable.

I would have expected the new revisions had a different app/istio label. If that is not the case we may need to rethink things. 

selectorLabels is defined in _helpers.tpl

**Comment by dblackdblack** [comment]
I manually set the value for the `app` label from 1.19 to match the value
from 1.16 (istio-ingressinternal).  That is so that I can do the traffic
cutover from one istio version to another by selecting for labels (in the
service object) that are the same between the two versions (the `app` and
`istio` labels) and then when I'm done, I add the `istio.io/rev=1-19-0`
selector on the service object.  i.e. it is important that for that use
case both the `istio` and the `app` label do not change between versions,
but that only the `istio.io/rev` label does.  I therefore need a way to
target `istio.io/rev` in the various object types (pdbs and deploys) beyond
what would normally be targeted by the service.

**Comment by dblackdblack** [comment]
The general idea is for version upgrades to make the service selector less
specific and target both the old and new versions of istio in order to gain
confidence.  Then add the version specific `istio.io/rev`
<http://istio.io/rev> selector when we have confidence.  In order for this
to work, of course, the other labels need to be identical, which is what
I've accomplished by setting the value of the `app` label to not contain
version information.

**Comment by dblackdblack** [comment]
The general idea is for version upgrades to make the service selector less
specific and target both the old and new versions of istio in order to gain
confidence.  Then add the version specific `istio.io/rev`
selector when we have confidence in the new version.  In order for this
to work, of course, the other labels need to be identical, which is what
I've accomplished by setting the value of the `app` label to not contain
version information.

Basically we want **only** `istio.io/rev` to be different between the old and new versions of istio.  We want `app` and `istio` labels to be static so that it is possible to target both the old and new versions with a single set of selectors.  When we're ready to pin to a version, we'll then add `istio.io/rev` as a selector.

**Comment by dblackdblack** [comment]
ping @howardjohn.  It would be great to support these additional helm fields.
